library/Booster/Pattern/Rewrite.hs

```diff
@@ -425,12 +437,12 @@ performRewrite def mLlvmLibrary mbMaxDepth cutLabels terminalLabels pat = do
 
     rewriteTrace t = do
         logRewrite $ pack $ renderDefault $ pretty t
-        modify $ \(counter, traces) -> (counter, traces |> t)
+        when doTracing $ modify $ \(counter, traces) -> (counter, traces |> t)
     incrementCounter = modify $ \(counter, traces) -> (counter + 1, traces)
 
     simplifyP :: Pattern -> StateT (Natural, Seq (RewriteTrace Pattern)) io Pattern
     simplifyP p = do
-        let result = evaluateTerm TopDown def mLlvmLibrary p.term
+        let result = evaluateTerm doTracing TopDown def mLlvmLibrary p.term
```

Comment:
We are losing the `-l Simplify` log if we do this... (see `logTraces` below).
Somewhat lame that one cannot find out whether that log level is active or not...
That problem goes away if we make `EquationM` a `MonadLoggerIO` instance (but that might be a bigger change)